[StyleCop] Fix all the warnings in TextMatchers

Aliandi opened the pull request:
- Added trailing commas
- Fixed spacing.
- Removed unnecessary parenthesis.
- Replaced UInt16 for built-in type alias ushort.
- Reordered properties and methods.
- Renamed private variable to lowercase.
- Reordered properties and methods.
- Changed field to private.

Batta32 committed:
Fix all StyleCop warnings in EnglishTimeZoneExtractorConfiguration

Batta32 committed:
Fix all StyleCop warnings in FrenchHolidayExtractorConfiguration

- Added spaces when needed

Aliandi committed:
Fix StyleCop warnings in TimexUtility

Aliandi committed:
Fix StyleCop warnings in BaseDateParserConfiguration

Commit:
Fix all StyleCop warnings in GermanHolidayExtractorConfiguration

Bill7zz committed:
Fix StyleCop warnings in BaseDateTimeParser

Bill7zz committed:
Fix StyleCop warnings in BaseTimeZoneParser

Batta32 committed:
Fix all StyleCop warnings in PortugueseCommonDateTimeParserConfiguration

- Addded spaces when needed

Batta32 committed:
Fix all StyleCop warnings in PortugueseDateTimeUtilityConfiguration

- Deleted unnecesary spaces
- Added newlines when needed

Batta32 committed:
Fix all StyleCop warnings in SpanishHolidayExtractorConfiguration

- Added spaces when needed

Aliandi committed:
Fix StyleCop warnings in TimeZoneUtility

Aliandi committed:
Fix StyleCop warnings in DataStructure

Aliandi committed:
Fix all StyleCop warnings in DateExtractor class -Fixed comments spacing -Added trailing commas

Aliandi committed:
Fix StyleCop warnings in DateExtractor

ceciliaavila committed:
Fix all StyleCop warnings in SimpleTokinizer and StringMatcher files.

- Fixed spacing.
- Removed unnecessary parenthesis.
- Replaced UInt16 for built-in type alias ushort.
- Reordered properties and methods.
- Renamed private variable to lowercase.

Bill7zz committed:
Fix StyleCop warnings in AbstractMatcher
-Remove trailing whitespace
-Set public method before private

Bill7zz committed:
Fix StyleCop warnings in Text\AcAutomaton
-Update Root variable to be property with private set
-Remove trailing whitespace

Bill7zz committed:
Fix StyleCop Warnings in Text\MatchResult
-Set constructor before properties

ceciliaavila committed:
Fix all StyleCop warnings in Token and TrieTree files.

- Fixed spacing.
- Reordered properties and methods.
- Changed field to private.

Aliandi committed:
Change tokenizer to a property

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/basetimezoneparser' into fis/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/basedatetimeparser' into fis/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/BaseDateParserConfiguration' into fis/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/DateExtractor' into fix/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/germanholidayextractorconfiguration' into fix/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/frenchholidayextractorconfiguration' into fix/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/englishtimezoneextractorconfiguration' into fix/stylecop/datetimeSpacingAndCommas

ceciliaavila committed:
Merge remote-tracking branch 'origin/fix/stylecop/DataStructure.cs' into fix/stylecop/datetimeSpacingAndCommas

Aliandi committed:
Merge branch 'fix/stylecop/portuguesecommondatetimeparserconfiguration' into fix/stylecop/datetimeSpacingAndCommas

Aliandi committed:
Merge branch 'fix/stylecop/portuguesedatetimeutilityconfiguration' into fix/stylecop/datetimeSpacingAndCommas

Aliandi committed:
Merge branch 'fix/stylecop/spanishholidayextractorconfiguration' into fix/stylecop/datetimeSpacingAndCommas

Aliandi committed:
Merge branch 'fix/stylecop/TimexUtility' into fix/stylecop/datetimeSpacingAndCommas

Aliandi committed:
Merge branch 'fix/stylecop/TimeZoneUtility' into fix/stylecop/datetimeSpacingAndCommas

Aliandi committed:
Merge branch 'fix/stylecop/abstractmatcher' into fix/stylecop/TextMatchers

Aliandi committed:
Merge branch 'fix/stylecop/acautomaton' into fix/stylecop/TextMatchers

Aliandi committed:
Merge branch 'fix/stylecop/matchresult' into fix/stylecop/TextMatchers

Aliandi committed:
Merge branch 'fix/stylecop/simpletokinizer' into fix/stylecop/TextMatchers

Aliandi committed:
Merge branch 'fix/stylecop/textmatchertoken' into fix/stylecop/TextMatchers

JuanAr commented on .NET/Microsoft.Recognizers.Text/Matcher/TrieTree.cs:
```diff
@@ -5,11 +5,10 @@ namespace Microsoft.Recognizers.Text.Matcher
 {
     public class TrieTree<T> : AbstractMatcher<T>
     {
-        protected readonly Node<T> root = new Node<T>();
```
Let's extract these changes (hiding a former protected field) to another PR and revert those in this branch.

JuanAr commented on .NET/Microsoft.Recognizers.Text/Matcher/AcAutomaton.cs:
```diff
@@ -5,16 +5,15 @@ namespace Microsoft.Recognizers.Text.Matcher
 {
     public class AcAutomaton<T> : AbstractMatcher<T>
     {
-        protected readonly AaNode<T> root = new AaNode<T>();
```
Let's extract this change to the same PR ad the one hiding the field. (They are similar changes, breaking changes because we are renaming a semi-public property: anyone with working code using this field in a child class will get an error)

JuanAr reviewed (changes requested):
I've left some feedback on some breaking changes: Changes that will somehow break existing functionality.
I asked to revert those changes from this PR, extract them to another PR and we'll think of a compromise solution and be clear about this win the PR.

Just remember, while fixing style warnings we should not change the existing behavior.

Aliandi committed:
Delete TrieTree.cs to create new branch with it

Aliandi committed:
Delete AcAutomaton.cs

Aliandi committed:
Fix all the StyleCop warnings in AcAutomaton and TrieTree
-Created a new branch to review breaking changes
-Change field from protected to private

Aliandi committed:
Adding files

Aliandi committed:
Change private field to protected and add setters and getters
